cart/src/main/java/com/yas/cart/service/CartService.java

```diff
@@ -43,43 +42,36 @@ public List<CartGetDetailVm> getCartDetailByCustomerId(String customerId) {
     }
 
     public CartGetDetailVm addToCart(List<CartItemVm> cartItemVms) {
-        if (CollectionUtils.isEmpty(cartItemVms)) {
-            throw new BadRequestException("Cart's item can't be null");
+        // Call API to check all products will be added to cart are existed
+        List<Long> productIds = cartItemVms.stream().map(CartItemVm::productId).toList();
+        List<ProductThumbnailVm> productThumbnailVmList = productService.getProducts(productIds);
+        if (productThumbnailVmList.size() != productIds.size()) {
+            throw new NotFoundException("Some products are not existed");
```

Comment:
Is this code necessary when we only get the valid product? moreover, if need to check then I think just call productRepository.existsById(productId) function to optimize performance.

Reply:
We can not call productRepository in cart service because database product is separate from cart service.
By the way, productService.getProducts(productIds) will make a request to product service and get all products by productIds attached.